api/handlers/listhandler.py

```diff
@@ -576,3 +584,132 @@ def packfile_end(self, cont_name, **kwargs):
         metadata = json.loads(self.request.GET.get('metadata'))
 
         return upload.process_upload(self.request, upload.Strategy.packfile, origin=self.origin, context={'token': token_id}, response=self.response, metadata=metadata)
+
+class AnalysesHandler(ListHandler):
+
+    def _check_ticket(self, ticket_id, _id, filename):
+        ticket = config.db.downloads.find_one({'_id': ticket_id})
+        if not ticket:
+            self.abort(404, 'no such ticket')
+        if ticket['ip'] != self.request.client_addr:
+            self.abort(400, 'ticket not for this source IP')
+        if not filename:
+            return self._check_ticket_for_batch(ticket)
+        if ticket.get('filename') != filename or ticket['target'] != _id:
+            self.abort(400, 'ticket not for this resource')
+        return ticket
+
+    def _check_ticket_for_batch(self, ticket):
+        if ticket.get('type') != 'batch':
+            self.abort(400, 'ticket not for this resource')
+        return ticket
+
+    def put(self, *args, **kwargs):
+        raise NotImplementedError("an analysis can't be modified")
```

Comment:
Generally Analyses should be considered immutable, but we already allow to add notes to them. For now, there is no other field mutable, but there could be, depending on how the [data model]( https://github.com/scitran/core/wiki/Data-Model,-v3#analysis-draft) will evolve.

If this is confusing I can change the Error type and add a more open error message.